### PR TITLE
Signup login fronend issue/21

### PR DESCRIPTION
**Pull request by mefaba**
### What this PR does (required):
- list PR functionality here

### Screenshots / Videos (required):
- 
image
image
image
image

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -16,18 +16,19 @@ function App(): JSX.Element {
...
-                <Route exact path="/signup" component={Signup} />
-                <Route exact path="/dashboard">
-                  <Dashboard />
-                </Route>
-                <Route path="*">
-                  <Redirect to="/login" />
-                </Route>
-              </Switch>
-            </SocketProvider>
+            {/* <SocketProvider> */}
```
Comment (moffatethan) [inline client/src/App.tsx:19]: We can just remove the socket provider if we don't need to use it

File: client/src/components/GoogleConnect/GoogleConnect.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                        fill="#EA4335"
+                        d="M -14.754 43.989 C -12.984 43.989 -11.404 44.599 -10.154 45.789 L -6.734 42.369 C -8.804 40.429 -11.514 39.239 -14.754 39.239 C -19.444 39.239 -23.494 41.939 -25.464 45.859 L -21.484 48.949 C -20.534 46.099 -17.884 43.989 -14.754 43.989 Z"
+                      />
+                    </g>
+                  </svg>
+                  Connect with Google
+                </Button>
+              </Link>
+            </Box>
+            {/* TODO */}
```
Comment (moffatethan) [inline client/src/components/GoogleConnect/GoogleConnect.tsx:58]: Todo what? These kind of comments are unnecessary.

File: client/src/pages/Login/Login.tsx [hunk 3]
```diff
@@ -1,59 +1,62 @@
...
-    { setSubmitting }: FormikHelpers<{ email: string; password: string }>,
-  ) => {
-    login(email, password).then((data) => {
+  const handleSubmit = ({ email }: { email: string }, { setSubmitting }: FormikHelpers<{ email: string }>) => {
+    checkUserEmail(email).then((data) => {
       if (data.error) {
+        //user email does not exists in database
         setSubmitting(false);
-        updateSnackBarMessage(data.error.message);
+        updateSnackBarMessage('User email does not  exists, plz try sign up');
```
Comment (moffatethan) [inline client/src/pages/Login/Login.tsx:24]: Abbreviations in front-end is not very good for a user to see.

File: client/src/App.tsx (hunk 1 above)
Comment (moffatethan) [inline client/src/App.tsx]: This can be removed

**Comment by moffatethan** [comment]
I remember reviewing this code already, please make sure we are following a good git flow so we don't see duplicated code pop up for reviewing. Thanks!

File: client/src/App.tsx (hunk 1 above)
Comment (mefaba) [reply client/src/App.tsx]: fix leftover comment and remove socketprovider a3f78bc

**Comment by moffatethan** [comment]
> I remember reviewing this code already, please make sure we are following a good git flow so we don't see duplicated code pop up for reviewing. Thanks!

Disregard

File: client/src/pages/SignUp/SignUp.tsx [hunk 4]
```diff
@@ -1,60 +1,60 @@
...
-        updateSnackBarMessage(data.error.message);
-      } else if (data.success) {
-        updateLoginContext(data.success);
-      } else {
-        // should not get here from backend but this catch is for an unknown issue
-        console.error({ data });
-
-        setSubmitting(false);
-        updateSnackBarMessage('An unexpected error occurred. Please try again');
+        updateSnackBarMessage('User email already exists, plz try log in');
```
Comment (moffatethan) [inline client/src/pages/SignUp/SignUp.tsx:24]: I would suggest in future utilizing proper spelling in our user errors.

**Review by moffatethan — APPROVED** [review]
Will approve, looks good to me. I think we need to reconsider the approach of the OAuth of typing their email then going through the Google OAuth. That isn't necessary. I'll approve for now and in future we can refactor accordingly.

**Pull request merged**